### DISCUSSION TITLE
[2.4] Notes and propositions

### DISCUSSION THREAD

**Discussion by kuraga** [blueman-project/blueman#2361] [Q&A]
blueman: 2.4.2
BlueZ: 5.76
Distribution: Calculate Linux (Gentoo-based)
Desktop environment: XFCE 4.18

* [x] I have consulted the [Troubleshooting page](https://github.com/blueman-project/blueman/wiki/Troubleshooting) and done my best effort to follow.

Dear developers, thanks for the release!
There are some questions and notes.

### ~p. 1.~

Are these battery level icons equal all the time?

![screenshot1a](https://github.com/blueman-project/blueman/assets/1063219/3d786add-e581-4c9a-9079-bc99c283ed17)
![screenshot1b](https://github.com/blueman-project/blueman/assets/1063219/acc6299f-b460-4745-8c48-149e03083ae8)

The actual battery level is 40%.
The first icon shows `green, 100%`.
The second shows `2/3`.

### p. 2.

![screenshot2](https://github.com/blueman-project/blueman/assets/1063219/946f4318-6d65-45c1-9728-85cc4b8664e3)

The ![image](https://github.com/blueman-project/blueman/assets/1063219/48231db7-b7d8-4c08-af90-4748bb6e50f5) icon should be 1px below.
Its height should be 1-2px less.

### p. 3.

![screenshot3](https://github.com/blueman-project/blueman/assets/1063219/5fc35b90-fab6-417c-b0a4-75af402f31d0)

Add the `100%-level` borders/background here?

### p. 4.

![screenshot4](https://github.com/blueman-project/blueman/assets/1063219/f1f2a3bf-6450-4101-acbf-c2befa3596c7)

Change this icon's (not the whole switch's) tooltip text to the _adapter name_?

### p. 5.

![screenshot5](https://github.com/blueman-project/blueman/assets/1063219/2e5aca6b-eabd-4f66-b900-07242126c012)

Add the `Turn off` item to the `Adapter` menu?

### ~p. 6.~

Don't whether this is Blueman or BlueZ but

![screenshot6](https://github.com/blueman-project/blueman/assets/1063219/7d4072fb-2159-4ddd-a5de-1b2699919c5a)

there is the general `HSP/HFP` item but there is no the `A2DP` one.

### p. 7.

Remove Bluetooth icon near `Off` in the `Profiles` menu?

![image](https://github.com/blueman-project/blueman/assets/1063219/8ad5496a-fa9d-4d4f-b73f-033913d6d7c6)

### ~p. 8.~

The Bluetooth devices window's title is `blueman-manager` instead of...
`Bluetooth devices`?
`Devices`?
`Blueman Manager`?

![image](https://github.com/blueman-project/blueman/assets/1063219/3b0fb3e9-21c4-40ba-8ab8-35334e7c1bf6)

### p. 9.

Add turn on/off toggle button on device item?

![image](https://github.com/blueman-project/blueman/assets/1063219/1c532dcf-6103-493b-82b5-e82dfb4d275f)

### p. 10.

Sort the device items list?

Thanks!

**Comment by cschramm** [comment]
> Are these battery level icons equal all the time?
> 
> ![screenshot1a](https://private-user-images.githubusercontent.com/1063219/319190258-3d786add-e581-4c9a-9079-bc99c283ed17.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTEzNDQsIm5iZiI6MTcxMjE1MTA0NCwicGF0aCI6Ii8xMDYzMjE5LzMxOTE5MDI1OC0zZDc4NmFkZC1lNTgxLTRjOWEtOTA3OS1iYzk5YzI4M2VkMTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDNUMTMzMDQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWVhZjhhZTk0NTAzMWNiMmM1MzBjNTdkMTg2ZWQ0YTY2YmYwOGYyZjY5YjkyOTNkYzE3ODQ3N2ZjMWEyYjczMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.PSxuhJ9R3Cm9tvhLA4kFxXifBWEscy6dPWuQUPciCWw) ![screenshot1b](https://private-user-images.githubusercontent.com/1063219/319190245-acc6299f-b460-4745-8c48-149e03083ae8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTEzNDQsIm5iZiI6MTcxMjE1MTA0NCwicGF0aCI6Ii8xMDYzMjE5LzMxOTE5MDI0NS1hY2M2Mjk5Zi1iNDYwLTQ3NDUtOGM0OC0xNDllMDMwODNhZTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDNUMTMzMDQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTlmM2FmNmI1MDRmOTk1ODczYjNjOTA4MWIxNDAzMWNiZjRhYjJlZDZjNTJiMzQ3NGMxZmNmYWRlOWE1ZjExMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.xwth3jzj5VQ6P0Xt1KS_K4uvSlA8n6XFMbDMp3M_TeI)
> 
> The actual battery level is 40%. The first icon shows `green, 100%`. The second shows `2/3`.

They are just generic batteries, yes. The first is the `U+1F50B` unicode character (from your fonts), the second is the `battery` icon (from your icon themes).

> _p. 6._
> 
> Don't whether this is Blueman or BlueZ but
> 
> ![screenshot6](https://private-user-images.githubusercontent.com/1063219/319195153-7d4072fb-2159-4ddd-a5de-1b2699919c5a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxNTEzNDQsIm5iZiI6MTcxMjE1MTA0NCwicGF0aCI6Ii8xMDYzMjE5LzMxOTE5NTE1My03ZDQwNzJmYi0yMTU5LTRkZGQtYTVkZS0xYjI2OTk5MTljNWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDNUMTMzMDQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWNlMWU1ODYwMWFkMGUzYjgwZjdlY2VjM2Y5OTdkZTFmOTVjZmI1ZjYwNzg4YzQwOGQ0YjIxNmZlZGNjMGEwNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.vKgYmIxe2cnAprxHPT0WIzM_6oh75RBZ7AjejfuCbHs)
> 
> there is the general `HSP/HFP` item but there is no the `A2DP` one.

It's not blueman and BlueZ is not involved. We just show a list of profiles provided via a pulseaudio API. PipeWire uses it to also offer codecs. Usually there *is* a generic A2DP entry as well, see https://github.com/blueman-project/blueman/issues/1860#issuecomment-1252923253.

**Comment by kuraga** [comment]
Updated for 2.4.2.
